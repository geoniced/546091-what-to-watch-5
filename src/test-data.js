export const noop = () => {};

export const movieCard = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  releaseDate: 2014,
};

export const filmListMock = [
  {
    backgroundColor: `#73B39A`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    id: 1,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    rating: 3.6,
    ratingsCount: 109661,
    releaseYear: 2008,
    runtime: 92,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    title: `Bronson`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    backgroundColor: `#795433`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    description: `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    director: `Quentin Tarantino`,
    genre: `Crime`,
    id: 2,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    rating: 1.5,
    ratingsCount: 1635992,
    releaseYear: 1994,
    runtime: 153,
    starring: [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    title: `Pulp Fiction`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    backgroundColor: `#828585`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Midnight_Special.jpg`,
    description: `A father and son go on the run, pursued by the government and a cult drawn to the child's special powers.`,
    director: `Jeff Nichols`,
    genre: `Action`,
    id: 3,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Midnight_Special.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/midnight-special.jpg`,
    rating: 3.3,
    ratingsCount: 67815,
    releaseYear: 2016,
    runtime: 112,
    starring: [`Michael Shannon`, `Joel Edgerton`, `Kirsten Dunst `],
    title: `Midnight Special`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    backgroundColor: `#B9B27E`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
    description: `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
    director: `Wachowski Brothers`,
    genre: `Action`,
    id: 4,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
    rating: 4.4,
    ratingsCount: 1503092,
    releaseYear: 1999,
    runtime: 136,
    starring: [`Keanu Reeves`, `Laurence Fishburne`, `Carrie-Anne Moss`],
    title: `Matrix`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    backgroundColor: `#E1DAD7`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/legend.jpg`,
    description: `Identical twin gangsters Ronald and Reginald Kray terrorize London during the 1960s.`,
    director: `Brian Helgeland`,
    genre: `Crime`,
    id: 5,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Legend.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/legend.jpg`,
    rating: 3.5,
    ratingsCount: 138487,
    releaseYear: 2015,
    runtime: 132,
    starring: [`Tom Hardy`, `Emily Browning`, `Taron Egerton`],
    title: `Legend`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    backgroundColor: `#A6B7AC`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/gangs_of_new_york.jpg`,
    description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    director: `Martin Scorsese`,
    genre: `Crime`,
    id: 6,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Gangs_of_New_York_Poster.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/gangs_of_new_york.jpg`,
    rating: 8.8,
    ratingsCount: 370881,
    releaseYear: 2002,
    runtime: 167,
    starring: [`Leonardo DiCaprio`, `Cameron Diaz`, `Daniel Day-Lewis`],
    title: `Gangs of new york`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    backgroundColor: `#C4C0C0`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/A_Star_is_Born.jpg`,
    description: `A musician helps a young singer find fame as age and alcoholism send his own career into a downward spiral.`,
    director: `Bradley Cooper`,
    genre: `Drama`,
    id: 7,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/A_Star_Is_Born.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/A_Star_Is_Born.jpg`,
    rating: 3.9,
    ratingsCount: 244484,
    releaseYear: 2018,
    runtime: 136,
    starring: [`Lady Gaga`, `Bradley Cooper`, `Sam Elliott`],
    title: `A Star Is Born`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    backgroundColor: `#EBC996`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/beach.jpg`,
    description: `Vicenarian Richard travels to Thailand and finds himself in possession of a strange map. Rumours state that it leads to a solitary beach paradise, a tropical bliss. Excited and intrigued, he sets out to find it.`,
    director: `Danny Boyle`,
    genre: `Adventure`,
    id: 8,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/beach.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/beach.jpg`,
    rating: 3.3,
    ratingsCount: 207824,
    releaseYear: 2000,
    runtime: 119,
    starring: [`Leonardo DiCaprio`, `Daniel York`, `Patcharawan Patarakijjanon`],
    title: `Beach`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    backgroundColor: `#D6CDAF`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Aviator.jpg`,
    description: `A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`,
    director: `Martin Scorsese`,
    genre: `Drama`,
    id: 9,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Aviator.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/aviator.jpg`,
    rating: 9.8,
    ratingsCount: 307174,
    releaseYear: 2014,
    runtime: 170,
    starring: [`Leonardo DiCaprio`, `Cate Blanchett`, `Kate Beckinsale`],
    title: `Aviator`,
    video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    videoPreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    backgroundColor: `#F0DBA2`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Johnny_English.jpg`,
    description: `After a sudden attack on the MI5, Johnny English, Britain's most confident yet unintelligent spy, becomes Britain's only spy.`,
    director: `Peter Howitt`,
    genre: `Comedy`,
    id: 10,
    isFavorite: false,
    posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Johnny_English.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/johnny-english.jpg`,
    rating: 10,
    ratingsCount: 136843,
    releaseYear: 2003,
    runtime: 88,
    starring: [`Rowan Atkinson`, `John Malkovich`, `Natalie Imbruglia`],
    title: `Johnny English`,
    video: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    videoPreview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  }
];
