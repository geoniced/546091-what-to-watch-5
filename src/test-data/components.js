import React from "react";
import PropTypes from "prop-types";

const MockComponentWithChildren = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponentWithChildren.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

export {MockComponentWithChildren};


const MockComponent = () => <div />;
export {MockComponent};
