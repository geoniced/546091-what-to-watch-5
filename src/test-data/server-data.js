export const filmsFromServer = [
  {
    "background_color": `#73B39A`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    "director": `Nicolas Winding Refn`,
    "genre": `Action`,
    "id": 1,
    "is_favorite": false,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    "rating": 3.6,
    "scores_count": 109661,
    "released": 2008,
    "run_time": 92,
    "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    "name": `Bronson`,
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    "background_color": `#795433`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    "director": `Quentin Tarantino`,
    "genre": `Crime`,
    "id": 2,
    "is_favorite": false,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    "rating": 1.5,
    "scores_count": 1635992,
    "released": 1994,
    "run_time": 153,
    "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    "name": `Pulp Fiction`,
    "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    "background_color": `#828585`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Midnight_Special.jpg`,
    "description": `A father and son go on the run, pursued by the government and a cult drawn to the child's special powers.`,
    "director": `Jeff Nichols`,
    "genre": `Action`,
    "id": 3,
    "is_favorite": false,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Midnight_Special.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/midnight-special.jpg`,
    "rating": 3.3,
    "scores_count": 67815,
    "released": 2016,
    "run_time": 112,
    "starring": [`Michael Shannon`, `Joel Edgerton`, `Kirsten Dunst `],
    "name": `Midnight Special`,
    "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    "background_color": `#B9B27E`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/matrix.jpg`,
    "description": `A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.`,
    "director": `Wachowski Brothers`,
    "genre": `Action`,
    "id": 4,
    "is_favorite": false,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/matrix.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/matrix.jpg`,
    "rating": 4.4,
    "scores_count": 1503092,
    "released": 1999,
    "run_time": 136,
    "starring": [`Keanu Reeves`, `Laurence Fishburne`, `Carrie-Anne Moss`],
    "name": `Matrix`,
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    "background_color": `#E1DAD7`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/legend.jpg`,
    "description": `Identical twin gangsters Ronald and Reginald Kray terrorize London during the 1960s.`,
    "director": `Brian Helgeland`,
    "genre": `Crime`,
    "id": 5,
    "is_favorite": false,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Legend.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/legend.jpg`,
    "rating": 3.5,
    "scores_count": 138487,
    "released": 2015,
    "run_time": 132,
    "starring": [`Tom Hardy`, `Emily Browning`, `Taron Egerton`],
    "name": `Legend`,
    "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    "background_color": `#A6B7AC`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/gangs_of_new_york.jpg`,
    "description": `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
    "director": `Martin Scorsese`,
    "genre": `Crime`,
    "id": 6,
    "is_favorite": false,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Gangs_of_New_York_Poster.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/gangs_of_new_york.jpg`,
    "rating": 8.8,
    "scores_count": 370881,
    "released": 2002,
    "run_time": 167,
    "starring": [`Leonardo DiCaprio`, `Cameron Diaz`, `Daniel Day-Lewis`],
    "name": `Gangs of new york`,
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    "background_color": `#C4C0C0`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/A_Star_is_Born.jpg`,
    "description": `A musician helps a young singer find fame as age and alcoholism send his own career into a downward spiral.`,
    "director": `Bradley Cooper`,
    "genre": `Drama`,
    "id": 7,
    "is_favorite": false,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/A_Star_Is_Born.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/A_Star_Is_Born.jpg`,
    "rating": 3.9,
    "scores_count": 244484,
    "released": 2018,
    "run_time": 136,
    "starring": [`Lady Gaga`, `Bradley Cooper`, `Sam Elliott`],
    "name": `A Star Is Born`,
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    "background_color": `#EBC996`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/beach.jpg`,
    "description": `Vicenarian Richard travels to Thailand and finds himself in possession of a strange map. Rumours state that it leads to a solitary beach paradise, a tropical bliss. Excited and intrigued, he sets out to find it.`,
    "director": `Danny Boyle`,
    "genre": `Adventure`,
    "id": 8,
    "is_favorite": false,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/beach.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/beach.jpg`,
    "rating": 3.3,
    "scores_count": 207824,
    "released": 2000,
    "run_time": 119,
    "starring": [`Leonardo DiCaprio`, `Daniel York`, `Patcharawan Patarakijjanon`],
    "name": `Beach`,
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    "background_color": `#D6CDAF`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Aviator.jpg`,
    "description": `A biopic depicting the early years of legendary Director and aviator Howard Hughes' career from the late 1920s to the mid 1940s.`,
    "director": `Martin Scorsese`,
    "genre": `Drama`,
    "id": 9,
    "is_favorite": false,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Aviator.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/aviator.jpg`,
    "rating": 9.8,
    "scores_count": 307174,
    "released": 2014,
    "run_time": 170,
    "starring": [`Leonardo DiCaprio`, `Cate Blanchett`, `Kate Beckinsale`],
    "name": `Aviator`,
    "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    "background_color": `#F0DBA2`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Johnny_English.jpg`,
    "description": `After a sudden attack on the MI5, Johnny English, Britain's most confident yet unintelligent spy, becomes Britain's only spy.`,
    "director": `Peter Howitt`,
    "genre": `Comedy`,
    "id": 10,
    "is_favorite": false,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Johnny_English.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/johnny-english.jpg`,
    "rating": 10,
    "scores_count": 136843,
    "released": 2003,
    "run_time": 88,
    "starring": [`Rowan Atkinson`, `John Malkovich`, `Natalie Imbruglia`],
    "name": `Johnny English`,
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  }
];
