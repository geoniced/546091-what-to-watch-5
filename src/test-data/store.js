import {NameSpace} from "../store/reducers/root-reducer";
import {AuthorizationStatus} from "../const";
import configureStore from "redux-mock-store";
import {filmListMock} from "./test-data";

const configuredStore = configureStore([]);
const mockedStore = configuredStore({
  [NameSpace.DATA]: {
    activeGenre: `All genres`,
    films: filmListMock,
    shownFilmsCount: 8,
    isLoading: false,
  },
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.AUTH,
  }
});

export {mockedStore};
