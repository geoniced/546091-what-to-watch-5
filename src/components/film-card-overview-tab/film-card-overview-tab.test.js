import React from "react";
import renderer from "react-test-renderer";
import {filmListMock} from "../../test-data/test-data";
import FilmCardOverviewTab from "./film-card-overview-tab";

describe(`FilmCardOverviewTab render`, () => {
  it(`renders FilmCardOverviewTab component with the first mock data piece`, () => {
    const tree = renderer
      .create(
          <FilmCardOverviewTab
            film={filmListMock[0]}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders FilmCardOverviewTab component with another data`, () => {
    const tree = renderer
      .create(
          <FilmCardOverviewTab
            film={filmListMock[5]}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

