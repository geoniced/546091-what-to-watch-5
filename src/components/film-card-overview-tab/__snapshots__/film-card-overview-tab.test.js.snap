// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCardOverviewTab render renders FilmCardOverviewTab component with another data 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.8
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        370881
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Martin Scorsese
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Leonardo DiCaprio, Cameron Diaz, Daniel Day-Lewis
      </strong>
    </p>
  </div>,
]
`;

exports[`FilmCardOverviewTab render renders FilmCardOverviewTab component with the first mock data piece 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      3.6
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Normal
      </span>
      <span
        className="movie-rating__count"
      >
        109661
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Nicolas Winding Refn
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Tom Hardy, Kelly Adams, Luing Andrews
      </strong>
    </p>
  </div>,
]
`;
